src/Runner/Runner.php

```diff
@@ -89,26 +123,250 @@ public function __construct(
+                $process = $processPool->getProcess($identifier);
+                $process->bindConnection($decoder, $encoder);
+                $fileChunk = $getFileChunk();
+
+                if (0 === \count($fileChunk)) {
+                    $process->request(['action' => ParallelAction::RUNNER_THANK_YOU]);
+                    $processPool->endProcessIfKnown($identifier);
+
+                    return;
+                }
+
+                $process->request(['action' => ParallelAction::RUNNER_REQUEST_ANALYSIS, 'files' => $fileChunk]);
+            });
+        });
+
+        $processesToSpawn = min(
+            $this->parallelConfig->getMaxProcesses(),
+            (int) ceil($this->fileCount / $this->parallelConfig->getFilesPerProcess())
         );
+        $processFactory = new ProcessFactory($this->input);
+
+        for ($i = 0; $i < $processesToSpawn; ++$i) {
+            $identifier = ProcessIdentifier::create();
+            $process = $processFactory->create(
+                $streamSelectLoop,
+                new RunnerConfig(
+                    $this->isDryRun,
+                    $this->stopOnViolation,
+                    $this->parallelConfig,
+                    $this->configFile
+                ),
+                $identifier,
+                $serverPort,
+            );
+            $processPool->addProcess($identifier, $process);
+            $process->start(
+                // [REACT] Handle workers' responses (multiple actions possible)
+                function (array $workerResponse) use ($processPool, $process, $identifier, $getFileChunk, &$changed): void {
+                    // File analysis result (we want close-to-realtime progress with frequent cache savings)
+                    if (ParallelAction::WORKER_RESULT === $workerResponse['action']) {
+                        $fileAbsolutePath = $workerResponse['file'];
+                        $fileRelativePath = $this->directory->getRelativePathTo($fileAbsolutePath);
+
+                        // Dispatch an event for each file processed and dispatch its status (required for progress output)
+                        $this->dispatchEvent(
+                            FixerFileProcessedEvent::NAME,
+                            new FixerFileProcessedEvent($workerResponse['status'])
+                        );
+
+                        if (isset($workerResponse['fileHash'])) {
+                            $this->cacheManager->setFileHash($fileRelativePath, $workerResponse['fileHash']);
+                        }
+
+                        // Pass-back information about applied changes (only if there are any)
+                        if (isset($workerResponse['fixInfo'])) {
+                            $changed[$fileRelativePath] = $workerResponse['fixInfo'];
+
+                            if ($this->stopOnViolation) {
+                                $processPool->endAll();
+
+                                return;
+                            }
```

Comment:
shouldn't we report `$workerResponse['errors']` first ?

Reply:
Moved it to the end 👍.